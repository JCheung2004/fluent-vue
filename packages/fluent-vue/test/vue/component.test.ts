import { createLocalVue, mount } from '@vue/test-utils'

import { FluentBundle, FluentResource } from '@fluent/bundle'
import ftl from '@fluent/dedent'

import FluentVue from '../../src'

describe('component', () => {
  let options: any
  let bundle: FluentBundle

  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(FluentVue)

    bundle = new FluentBundle('en-US')

    const fluent = new FluentVue({
      bundles: [bundle]
    })

    options = {
      fluent,
      localVue
    }
  })

  it('handles simple translations', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      key = Inner data
      `)
    )

    const component = {
      template: '<i18n path="key"></i18n>'
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('interpolates components', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      key = Inner data {$child} test
      `)
    )

    const component = {
      template: `
        <i18n path="key">
          <template #child>
            <b>Inner text</b>
          </template>
        </i18n>`
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })

  it('can accept parameters', () => {
    // Arrange
    bundle.addResource(
      new FluentResource(ftl`
      key = Hello {$name} {$child} test
      `)
    )

    const component = {
      data() {
        return {
          name: 'John'
        }
      },
      template: `
        <i18n path="key" :data="{ name }">
          <template #child>
            <b>Inner text</b>
          </template>
        </i18n>`
    }

    // Act
    const mounted = mount(component, options)

    // Assert
    expect(mounted).toMatchSnapshot()
  })
})
