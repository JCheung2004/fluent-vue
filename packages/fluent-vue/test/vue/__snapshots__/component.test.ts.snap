// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component can accept parameters 1`] = `<span>Hello ⁨John⁩ ⁨<b>Inner text</b>⁩ test</span>`;

exports[`component handles simple translations 1`] = `<span>Inner data</span>`;

exports[`component interpolates components 1`] = `<span>Inner data ⁨<b>Inner text</b>⁩ test</span>`;
